src/Elastic.Clients.Elasticsearch/Types/Core/Search/SuggestDictionary.cs

```diff
@@ -0,0 +1,215 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+using Elastic.Clients.Elasticsearch.Serialization;
+
+namespace Elastic.Clients.Elasticsearch.Core.Search;
+
+[GenericConverter(typeof(SuggestDictionaryConverter<>), unwrap:true)]
+public sealed partial class SuggestDictionary<TDocument> :
+	IsAReadOnlyDictionary<string, IReadOnlyCollection<ISuggest>>
+{
+	public SuggestDictionary(IReadOnlyDictionary<string, IReadOnlyCollection<ISuggest>> backingDictionary) :
```

Comment:
Could we make this ctor internal as we only expect to deserialize into this type ourselves and create it from the converter?

Reply:
The code generator seems to generate most constructors as `public` at the moment. I tried to keep this hand-crafted class as close as possible to the generator output in order to avoid breaking changes in the future like we discussed. We could potentially improve the generator for outputting `protected/internal` constructors for types that are exclusively used responses.

Reply:
Also true. For some of these types used only on responses, it shouldn't generate them with public ctors. We try to determine which types are used where in the generator so we can make response-only types read-only and init-only. I probably missed that in some places for these special cases. Personally, I'd start with it internal as making it public later if that happens in code-gen is non-breaking.

Reply:
Sounds good! Will change it to `internal` for now.